### ISSUE TITLE
'Freeform' block is not supported error appears in between a custom post type's template blocks 

### ISSUE THREAD

**Issue by skorasaurus** [WordPress/gutenberg#52811]
### Description

After the introduction of https://github.com/WordPress/gutenberg/pull/52325, 

I'm running into a unique case specifically: when editing already published custom posts whose content is inserted through the register_post_type($template), A block warning appears in the editor stating 'Your site doesn’t include support for the "core/freeform" block. You can leave this block intact or remove it entirely.' 

As far as I know, [the block content I have inserted](https://gitlab.com/cpl/test-case-52811/-/blame/main/register.php#L33) into the custom post type is valid and does not contain a classic block. 

**Notably, I am unable to reproduce this in Gutenberg 16.2.1;** 

### Step-by-step reproduction instructions

1. [Install Gutenberg nightly](https://gutenbergtimes.com/need-a-zip-from-master/#nightly) from or update your current branch to trunk (and build)
2a. Go to Experiments http://yoursite.test/wp-admin/admin.php?page=gutenberg-experiments.
2b. Enable the "Disable TinyMCE and Classic block" experiment and click "Save Changes".
3. Install and activate the test case plugin - (https://gitlab.com/cpl/test-case-52811)
4. create new 'agenda' (the new custom post type); 
5. publish the post 
6. refresh the editor in your browser 

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/955351/12421e0e-a392-4467-a547-0f86d352245a

### Environment info

WP 6.2; Gutenberg built from trunk 2023-07-18 or newer. Have been able to reproduce with and without classic editor plugin installed. 

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

yes (except the test case plugin)

**Comment by Mamaduka** [comment]
Cc @tyxla

**Comment by tyxla** [comment]
Thank you @Mamaduka for the ping and @skorasaurus for the thorough report! 

I'll work on a fix this week.

**Comment by tyxla** [comment]
Alright, I've got a few related and direct fixes of this one - could use some 👀 if you folks have a chance:

* https://github.com/WordPress/gutenberg/pull/52936 - sets the freeform content handler only if the block exists, part of the fix for this issue.
* https://github.com/WordPress/gutenberg/pull/52937 - stops displaying missing blocks for empty freeform blocks if that block is not registered.
* https://github.com/WordPress/gutenberg/pull/52938 - handles existing `core/freeform` blocks in post content properly
* https://github.com/WordPress/gutenberg/pull/52935 - resolves some warning during heartbeat XHR requests and on the page to create a post from custom post types

Let me know what you think!

**Comment by tyxla** [comment]
@skorasaurus now that we merged https://github.com/WordPress/gutenberg/pull/52935 and https://github.com/WordPress/gutenberg/pull/52936, would you mind giving this a try with the latest `trunk`?

~We still have #52938 open, but that resolves a different scenario, so you should be good when testing with the above steps.~ Actually, this landed as well already. 

Let me know how it goes, thanks 🙌

**Comment by skorasaurus** [comment]
Thank you again for your work @tyxla and @Mamaduka (and anyone who reviewed the PRs). 

Using the latest trunk, I am unable to reproduce the issue [described earlier](https://github.com/WordPress/gutenberg/issues/52811#issue-1814787270). I tested using my reduced test-case plugin and in another plugin that I maintain that has more complex blocks in the custom post template.  

I edited and updated some other classic blocks on other pages on my site(s), and they worked as expected as well.
